tests/system.rs

```diff
@@ -66,27 +84,58 @@ fn print_mismatches(result: HashMap<String, String>) {
 static HANDLE_RESULT: &'static Fn(HashMap<String, String>) = &handle_result;
 
 pub fn idempotent_check(filename: String) -> Result<(), HashMap<String, String>> {
+    let config = get_config(&filename);
+
+    println!("config: {}", &config);
+
     let args = vec!["rustfmt".to_owned(), filename];
-    let mut def_config_file = fs::File::open("default.toml").unwrap();
-    let mut def_config = String::new();
-    def_config_file.read_to_string(&mut def_config).unwrap();
     // this thread is not used for concurrency, but rather to workaround the issue that the passed
     // function handle needs to have static lifetime. Instead of using a global RefCell, we use
     // panic to return a result in case of failure. This has the advantage of smoothing the road to
     // multithreaded rustfmt
     thread::catch_panic(move || {
-        run(args, WriteMode::Return(HANDLE_RESULT), &def_config);
+        run(args, WriteMode::Return(HANDLE_RESULT), &config);
     }).map_err(|any|
         *any.downcast().unwrap()
     )
 }
 
+// Reads test config file from comments and loads it
+fn get_config(file_name: &str) -> String {
+    let config_file_name = read_significant_comment(file_name, "// rustfmt-config: ")
```

Comment:
Doesn't need to be done for this PR, but it would be nice if we didn't have to supply the `// ` or the `: ` to `read_significant_comment`, so we could parse something like:
```
//    rustfmt-config:foo.toml
```

Also, we should probably only look for these comments until we hit the first non-significant-comment, non-whitespace line, rather than in the first 5 lines.

Doesn't need to happen here, but if you could file issues or add a FIXME (or both), that would be good.
```